paddle/framework/operator.cc

```diff
@@ -79,6 +79,10 @@ std::vector<std::string> OperatorBase::Outputs(const std::string& name) const {
       outputs_.begin() + output_format.at(offset + 1)};
 }
 
+void OperatorBase::InferShape(const std::shared_ptr<Scope>& scope) const {
+  InferShapeImpl(InferShapeContext(this, scope));
```

Comment:
It cannot be done, because inside `InferShape`/`Run` in some operator, e.g., `RNN`, the developer will create a new local `Scope` which uses `std::shared_ptr<Scope>` as an argument.